pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -453,11 +454,11 @@ func (dh *dnsHelper) getDNSHealthCheckProbes(ctx context.Context, gateway *gatew
 	})
 }
 
-func getProbesForEndpoint(endpoint *v1alpha1.Endpoint, probes []*v1alpha1.DNSHealthCheckProbe) []*v1alpha1.DNSHealthCheckProbe {
+func getProbesForEndpoint(endpoint *v1alpha1.Endpoint, probes []*v1alpha1.DNSHealthCheckProbe, listenerName string) []*v1alpha1.DNSHealthCheckProbe {
 	retProbes := []*v1alpha1.DNSHealthCheckProbe{}
 	for _, probe := range probes {
 		for _, target := range endpoint.Targets {
-			if strings.Contains(probe.Name, target) {
+			if strings.Contains(probe.Name, target) && strings.Contains(probe.Name, listenerName) {
```

Comment:
This looks good, but I wonder if it'd be a good idea to externalise this logic to a separate function, same with the logic that creates the name of the probe when the probes are created. That way if it changes it would be easier to remember to make sure that they match. Wdyt?

Reply:
Yeah. For sure I can do that. One other thing I considered was adding the listener name to the labels. We have them in annotations but these don't offer the same functionality as labels when it comes to LabelSelector for ListOptions.

Reply:
Yeah I think it was me who added these as annotations instead of labels and I kinda regret it

Reply:
I've kept using the name rather than a rewrite on the annotations/labels for now. Ready for review again if you have some time. 🙏